react/features/base/jwt/middleware.js

```diff
@@ -183,11 +282,24 @@ function _setJWT(store, next, action) {
             action.issuer = iss;
             if (context) {
                 action.callee = context.callee;
-                action.caller = context.user;
+                action.caller = _normalizeCallerFields(context.user);
                 action.group = context.group;
                 action.server = context.server;
+
+                if (action.caller) {
+                    _overwriteLocalParticipant(store, action.caller);
+                }
             }
         }
+    } else if (!jwt && !Object.keys(actionPayload).length) {
```

Comment:
I'm taking care of it but for the record: I'm not in support of `if` + `else if` constructs checking one and the same condition and/or its reverse. For example, here we now have:

````javascript
if (jwt && !Object.keys(actionPayload).length)
else if (!jwt && !Object.keys(actionPayload).length)
````

1. `!Object.keys(actionPayload).length` is a common condition so it should be checked once. Additionally, it's not even a trivial/cheap check.
2. `if` checks that `jwt` is truthy, so the else of `else if` already guarantees that `jwt` is falsy.

Reply:
well I wasn't able to understand why we even check for action payload at all...

Reply:
is it like it could be called twice for SET_JWT ? or something else extracted the fields already ?